erpnext/crm/doctype/opportunity/opportunity.py

```diff
@@ -628,14 +629,48 @@ def get_customer_from_opportunity(source):
 
 
 @frappe.whitelist()
-def submit_communication(name, contact_date, remarks, submit_follow_up=False):
+def schedule_follow_up(name, schedule_date, to_discuss=None):
+	if not schedule_date:
+		frappe.throw(_("Schedule Date is mandatory"))
+
+	schedule_date = getdate(schedule_date)
+
+	if schedule_date < getdate():
+		frappe.throw(_("Can't schedule a follow up for past dates"))
+
+	opp = frappe.get_doc("Opportunity", name)
+	dup = [d for d in opp.get('contact_schedule') if d.get('schedule_date') == schedule_date]
+
+	if dup:
+		dup = dup[0]
+		if (dup.to_discuss and to_discuss != dup.to_discuss) or (not dup.to_discuss and not to_discuss):
+			frappe.throw(_("Row #{0}: Follow Up already scheduled for {1}".format(dup.idx, frappe.format(dup.schedule_date))))
+		else:
+			dup.to_discuss = to_discuss
+	else:
+		opp.append('contact_schedule', {
+			'schedule_date': schedule_date,
+			'to_discuss': to_discuss
+		})
+
+	opp.save()
+
+
+@frappe.whitelist()
+def submit_communication(opportunity, contact_date, remarks, submit_follow_up=False,
+		maintenance_schedule=None, maintenance_schedule_row=None):
 	if not remarks:
 		frappe.throw(_('Remarks are mandatory for Communication'))
 
-	if not frappe.db.exists('Opportunity', name):
-		frappe.throw(_("Opportunity does not exist"))
+	remarks = clean_whitespace(remarks)
 
-	opp = frappe.get_doc('Opportunity', name)
+	if frappe.db.exists('Opportunity', opportunity):
+		opp = frappe.get_doc('Opportunity', opportunity)
+	elif maintenance_schedule and maintenance_schedule_row:
+		opp = create_maintenance_opportunity(maintenance_schedule, maintenance_schedule_row)
```

Comment:
What if Opportunity already exists against `maintenance_schedule` and `maintenance_schedule_row`